services/data-info/src/data_info/services/write.clj

```diff
@@ -0,0 +1,59 @@
+(ns data-info.services.write
+  (:use [clj-jargon.init :only [with-jargon]])
+  (:require [clojure-commons.file-utils :as ft]
+            [clojure-commons.error-codes :as ce]
+            [clj-jargon.item-info :as info]
+            [clj-jargon.item-ops :as ops]
+            [ring.middleware.multipart-params :as multipart]
+            [data-info.services.stat :as stat]
+            [data-info.util.config :as cfg]
+            [data-info.util.validators :as validators]))
+
+(defn- save-file-contents
+  "Save an istream to a destination. Relies on upstream functions to validate."
+  [cm istream user dest-path]
+  (ops/copy-stream cm istream user dest-path)
+  dest-path)
+
+(defn- create-at-path
+  "Create a new file at dest-path from istream.
+
+   Error if the path exists or if the destination directory does not exist or is not writeable."
+  [cm istream user dest-path]
+  (let [dest-dir (ft/dirname dest-path)]
+    (validators/user-exists cm user)
+    (validators/path-not-exists cm dest-path)
+    (validators/path-exists cm dest-dir)
+    (validators/path-writeable cm user dest-dir)
+    (save-file-contents cm istream user dest-path)))
+
+(defn- overwrite-path
```

Comment:
It appears that this function is never called. I assume that it will be called at some point in the future, but I thought I should point it out in case that assumption is incorrect.